xds/src/main/java/io/grpc/xds/EdsLoadBalancer.java

```diff
@@ -91,29 +92,40 @@
       LocalityStoreFactory localityStoreFactory,
       Bootstrapper bootstrapper,
       XdsChannelFactory channelFactory) {
-    this.edsLbHelper = edsLbHelper;
-    this.channelLogger = edsLbHelper.getChannelLogger();
-    this.resourceUpdateCallback = resourceUpdateCallback;
-    this.lbRegistry = lbRegistry;
-    this.localityStoreFactory = localityStoreFactory;
+    this.edsLbHelper = checkNotNull(edsLbHelper, "edsLbHelper");
+    this.resourceUpdateCallback = checkNotNull(resourceUpdateCallback, "resourceUpdateCallback");
+    this.lbRegistry = checkNotNull(lbRegistry, "lbRegistry");
+    this.localityStoreFactory = checkNotNull(localityStoreFactory, "localityStoreFactory");
+    this.bootstrapper = checkNotNull(bootstrapper, "bootstrapper");
+    this.channelFactory = checkNotNull(channelFactory, "channelFactory");
     this.switchingLoadBalancer = new GracefulSwitchLoadBalancer(edsLbHelper);
-    this.bootstrapper = bootstrapper;
-    this.channelFactory = channelFactory;
+    logId = InternalLogId.allocate("eds-lb", edsLbHelper.getAuthority());
+    logger = XdsLogger.withLogId(logId);
+    logger.log(XdsLogLevel.INFO, "Created");
   }
 
   @Override
   public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
-    channelLogger.log(ChannelLogLevel.DEBUG, "Received ResolvedAddresses {0}", resolvedAddresses);
-
+    logger.log(XdsLogLevel.DEBUG, "Received resolution result: {0}", resolvedAddresses);
     Object lbConfig = resolvedAddresses.getLoadBalancingPolicyConfig();
-    if (!(lbConfig instanceof XdsConfig)) {
+    if (lbConfig == null) {
```

Comment:
checkNotNull()?

Reply:
Since we have EDS-only use case, we should be more tolerant for `ResolvedAddesses` received externally. For CDS policy, it's always the `XdsNameResolver` gives us the resolution result, so we assert it should be guaranteed to be there. Otherwise, it's our bug. But for EDS policy, in EDS-only mode we get resolution result from `DnsNameResolver`, which doesn't guarantee a load balancing config. If EDS policy does not receive the EDS config, it could be the remote system's issue. We should just handle that gracefully.

Reply:
DnsNameResolver will parse the config while selecting lb policy. If it's not EDS config when EDS policy is selected it will handleResolutionError.